cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -680,43 +680,100 @@ public boolean apply(RunRecordMeta runRecordMeta) {
   }
 
   public void upgradeVersionKeys() {
```

Comment:
It would be good to not do all of these in same transaction, we can have separate transactions for each.